File: Comunication Circuits/dr mirrezaie/code/Linville_Stability.py
import numpy as np

row = column = 2 
print("enter admittance matrix(Ytr): ")
  
entries = list(map(str, input().split()))

Ytr = np.array(entries, dtype=complex).reshape(row, column)

C = (abs(Ytr[1, 0] * Ytr[0, 1])) / (2 * Ytr[0, 0].real * Ytr[1, 1].real - (Ytr[1, 0] * Ytr[0, 1]).real)
print('\n\nC = %f\n\n' %C)

if(Ytr[0, 0].real > 0 and Ytr[1, 1].real > 0 and (C >= 0 and C <= 1)):
    print('stable')
else:
    print('unstable')

#2.7+6.8j 0-0.5j 53-22j 0.1+1.5j

File: Comunication Circuits/Dr MirRezaie/Code/Linville_Stability.m
clear;clc;close all;

Ytr = input('enter admittance matrix(Ytr): ');
C = (abs(Ytr(1, 2)*Ytr(2, 1))) / (2*real(Ytr(1, 1))*real(Ytr(2, 2)) - real(Ytr(1, 2)*Ytr(2, 1))) ;
fprintf('\nC = %f\n\n', C);
if( (real(Ytr(1, 1)) > 0) && (real(Ytr(2, 2)) > 0) &&  (C >= 0 && C <= 1) )
    disp('stable');
else
    disp('unstable');
end


%[2.7+6.8j 0-0.5j ; 53-22j 0.1+1.5j]